src/markdown.c

```diff
@@ -769,8 +769,33 @@ char_entity(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t max_
 		if (!is_valid_numeric_entity(entity_val))
 			return 0;
 	} else {
-		if (!is_allowed_named_entity((const char *)data, end))
+		/* Lookup the entity in the named entity table. */
+		const struct html_entity *entity = is_allowed_named_entity((const char *)data, end);
+		if (!entity)
 			return 0;
+		if (entity->codepoints[0]) {
+			/* Convert the entity to numeric entities. */
+			size_t entitystr_size = MAX_NUM_ENTITY_LEN+10;
```

Comment:
I worked it out when I wrote the code, it's supposed to be `+5` (`&#x;` + NULL). But I'm never confident in my arithmetic when I do that, so I always bump it up a little more. I'll put it back down to 5.